pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -416,6 +418,37 @@ func (s *DiscoveryServer) generateEndpoints(
 	return l
 }
 
+// EdsGenerator implements the new Generate method for EDS, using the in-memory, optimized endpoint
+// storage in DiscoveryServer.
+type EdsGenerator struct {
+	Server *DiscoveryServer
+}
+
+func (eds *EdsGenerator) Generate(proxy *model.Proxy, push *model.PushContext, w *model.WatchedResource, updates model.XdsUpdates) model.Resources {
+	resp := []*any.Any{}
+	edsUpdatedServices := model.ConfigNamesOfKind(updates, model.ServiceEntryKind)
+	// All clusters that this endpoint is watching. For 1.0 - it's typically all clusters in the mesh.
+	// For 1.1+Sidecar - it's the small set of explicitly imported clusters, using the isolated DestinationRules
+	for _, clusterName := range w.ResourceNames {
+
+		l := eds.Server.generateEndpoints(clusterName, proxy, push, edsUpdatedServices)
+		if l == nil {
+			continue
+		}
+		bany, err := types.MarshalAny(l)
```

Comment:
this is `gogo.marshal` on XDS - I don't think it will work properly. This should all be golang

Reply:
XDS protocol is defined as 'Any' - on the wire it must be in protocol buf format, which is well defined. It doesn't matter what you use to marshal - gogo or golang or writing bytes by hand. 

It seems to work properly - and it should, that's the point of protobuf stable wire format, any language and implementation will work.

Reply:
You are calling `gogo.Marshal(golang type)` though. It should be `golang.marshal(golang type)`, right? I don't really see how it works

Reply:
types.MarshalAny is used with golang - i.e. go-contro-plane types, like EDS.

gogo is used for K8S and Istio types.

Reply:
The grpc test is using this code - and seems to be passing now, i.e. they get the endpoint and can connect.